### DISCUSSION TITLE
How much data can be stored?

### DISCUSSION THREAD

**Discussion by hajindersingh645** [dexie/Dexie.js#1341] [Q&A] [answered]
Hi,
I am looking for a solution / library which save offline data of products upto 10,000 rows or more. Just wanted to confirm how much data storage is available or supported by this library.

Thank you

**Comment by dfahlander** [comment, marked as answer]
Dexie has no more limit than what the browser/device allows. Indexeddb has no row limit but there can be storage limits on the device depending on browser software, device type and available hard disk on the device. [StorageManager api](https://dexie.org/docs/StorageManager) can be used to see storage quota.
